cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -4,11 +4,51 @@
 package integration
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+
 	"github.com/jaegertracing/jaeger/plugin/storage/integration"
 )
 
+func createConfigWithEncoding(t *testing.T, configFile string) string {
+	data, err := os.ReadFile(configFile)
+	require.NoError(t, err, "Failed to read config file: %s", configFile)
+
+	var config map[string]any
+	err = yaml.Unmarshal(data, &config)
+	require.NoError(t, err, "Failed to unmarshal YAML data from config file: %s", configFile)
+
+	// Function to recursively search and replace the encoding
+	var replaceEncoding func(m map[string]any)
+	replaceEncoding = func(m map[string]any) {
```

Comment:
```suggestion
	replaceEncoding := func(m map[string]any) {
```

Reply:
The` replaceEncoding` function is being referenced within itself, but when we declare the function using the shorthand approach (:=), it hasn't been fully defined at that point so wouldn't that be throwing up an error saying it's undefined?